corehq/apps/hqadmin/management/commands/stale_data_in_es.py

```diff
@@ -154,19 +164,47 @@ class CaseHelper:
 
     @classmethod
     def run(cls, run_config):
-        for chunk in cls.get_sql_chunks(run_config):
-            yield from cls._yield_missing_in_es(chunk)
+        all_domains_with_search_index = []
+        all_domains_with_dedicated_search_index = []
 
-    @staticmethod
-    def get_sql_chunks(run_config):
-        domain = run_config.domain if run_config.domain is not ALL_DOMAINS else None
+        if run_config.domain is ALL_DOMAINS:
+            all_domains = None
+            all_domains_with_search_index = \
+                [domain for domain in Domain.get_all_names() if domain_needs_search_index(domain)]
+            all_domains_with_dedicated_search_index = \
+                [domain for domain in Domain.get_all_names() if domain in settings.CASE_SEARCH_SUB_INDICES]
+        else:
+            all_domains = run_config.domain
+            if domain_needs_search_index(run_config.domain.name):
+                all_domains_with_search_index = [run_config.domain.name]
+            if run_config.domain in settings.Case_SEARCH_SUB_INDICES:
+                all_domains_with_dedicated_search_index = [run_config.domain]
+
+        case_query = CaseES(for_export=True)
+        for chunk in cls.get_sql_chunks(all_domains, run_config):
+            yield from cls._yield_missing_in_es(chunk, case_query)
+
+        # USH-4684: The yield from these for loop can be moved in the one for CaseES
+        case_search_query = CaseSearchES(for_export=True)
+        for domain in all_domains_with_search_index:
+            for chunk in cls.get_sql_chunks(domain, run_config):
+                yield from cls._yield_missing_in_es(chunk, case_search_query)
+
+        for domain in all_domains_with_dedicated_search_index:
+            case_search_index = settings.CASE_SEARCH_SUB_INDICES[domain]['index_cname']
+            case_search_query = CaseSearchES(for_export=True, index=case_search_index)
+            for chunk in cls.get_sql_chunks(domain, run_config):
+                yield from cls._yield_missing_in_es(chunk, case_search_query)
 
+    @staticmethod
+    def get_sql_chunks(domain, run_config, domains_with_dedicated_index=None):
```

Comment:
`domains_with_dedicated_index` is unused now, right?

Reply:
correct. Removed.